bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/internal/events/OSGiEventManager.java

```diff
@@ -271,36 +197,52 @@ public Void run() throws Exception {
 
     private void assertValidArgument(Event event) throws IllegalArgumentException {
         String errorMsg = "The %s of the 'event' argument must not be null or empty.";
-        Preconditions.checkArgument(event != null, "Argument 'event' must not be null.");
-        Preconditions.checkArgument(event.getType() != null && !event.getType().isEmpty(),
-                String.format(errorMsg, "type"));
-        Preconditions.checkArgument(event.getPayload() != null && !event.getPayload().isEmpty(),
-                String.format(errorMsg, "payload"));
-        Preconditions.checkArgument(event.getTopic() != null && !event.getTopic().isEmpty(),
-                String.format(errorMsg, "topic"));
+        String value;
+
+        if (event == null) {
+            throw new IllegalArgumentException("Argument 'event' must not be null.");
+        }
+        if ((value = event.getType()) == null || value.isEmpty()) {
+            throw new IllegalArgumentException(String.format(errorMsg, "type"));
+        }
+        if ((value = event.getPayload()) == null || value.isEmpty()) {
+            throw new IllegalArgumentException(String.format(errorMsg, "payload"));
+        }
+        if ((value = event.getTopic()) == null || value.isEmpty()) {
+            throw new IllegalArgumentException(String.format(errorMsg, "topic"));
+        }
     }
 
     private void assertValidState(EventAdmin eventAdmin) throws IllegalStateException {
-        Preconditions.checkArgument(eventAdmin != null, "The event bus module is not available!");
+        if (eventAdmin == null) {
+            throw new IllegalArgumentException("The event bus module is not available!");
```

Comment:
This should be an `IllegalStateException`.

Reply:
Fixed.